### PR TITLE
Feat: Clusteradm init --wait for progressive hub init

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

**Comment by yue9944882** [comment]
/hold

until https://github.com/open-cluster-management-io/clusteradm/pull/93, will rebase after the precedence merged

File: pkg/cmd/init/exec.go [hunk 1]
```diff
@@ -140,10 +149,29 @@ func (o *Options) run() error {
...
+			return err
+		}
+	}
+
 	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
 	if err != nil {
 		return nil
 	}
+	if len(o.outputJoinCommandFile) > 0 {
+		sh, err := os.OpenFile(o.outputJoinCommandFile, os.O_CREATE|os.O_WRONLY, 0755)
```
Comment (qiujian16) [inline pkg/cmd/init/exec.go:165]: Should have a flag to only output this command so we can pipe it to a file?

Comment (yue9944882) [reply pkg/cmd/init/exec.go]: are you thinking about utilizing the output from init e.g.?

> echo "$JOIN_CLUSTER_NAME" | cluster init --cmd-only --pipe-stdin

File: pkg/cmd/join/exec.go [hunk 2]
```diff
@@ -193,7 +193,7 @@ func waitUntilRegistrationOperatorConditionIsTrue(f util.Factory, timeout int64)
 	operatorSpinner.Start()
 	defer operatorSpinner.Stop()
 
-	return waitUntilConditionIsTrue(
+	return helpers.WatchUntil(
```
Comment (qiujian16) [inline pkg/cmd/join/exec.go:196]: Followup: We should use "degraded" condition of klusterlet here instead of reading pod status.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**